### ISSUE TITLE
[ReflectionUtils] - ReflectionUtils is unable to find NMS classes on 1.17 due to NMS repackaging

### ISSUE THREAD

**Issue by LOOHP** [CryptoMorin/XSeries#123]
**Description**
1.17 package classes differently, field names and enum names are also back to their obfuscated form. This causes ReflectionUtils unable to be initialized properly. This causes some utilities such as `SkullUtils` to be unable to function (as ReflectionUtils is not Initialized).

Example:
`net.minecraft.server.VERSION.ItemStack` is now `net.minecraft.world.item.ItemStack`
`playerConnection` field in `EntityPlayer` is now `b`

Errors: (Thought I'd include them anyway)
```
java.lang.ClassNotFoundException: net.minecraft.server.v1_17_R1.EntityPlayer
         at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:142)
         at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:99)
         at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:519)
         at java.base/java.lang.Class.forName0(Native Method)
         at java.base/java.lang.Class.forName(Class.java:375)
         at com.loohp.interactivechat.libs.com.cryptomorin.xseries.ReflectionUtils.getNMSClass(ReflectionUtils.java:121)
         at com.loohp.interactivechat.libs.com.cryptomorin.xseries.ReflectionUtils.<clinit>(ReflectionUtils.java:87)
         at com.loohp.interactivechat.libs.com.cryptomorin.xseries.SkullUtils.<clinit>(SkullUtils.java:69)
```
_Causing SkullUtils to fail_
```
java.lang.ExceptionInInitializerError: null
        at com.loohp.interactivechat.libs.com.cryptomorin.xseries.SkullUtils.<clinit>(SkullUtils.java:69) ~[?:?]
```

**Version**
1.17

**Comment by SirLeezus** [comment]
I've been trying to help update this, I was able to resolve all the errors with the following:
`public static final String
            VERSION = Bukkit.getServer().getClass().getPackage().getName().split("\\.")[3],
            CRAFTBUKKIT = "org.bukkit.craftbukkit." + VERSION + '.',
            NMS = "net.minecraft.";

    private static final MethodHandle PLAYER_CONNECTION;

    private static final MethodHandle GET_HANDLE;

    private static final MethodHandle SEND_PACKET;

    static {
        Class<?> entityPlayer = getNMSClass("server.level.EntityPlayer");
        Class<?> craftPlayer = getCraftClass("entity.CraftPlayer");
        Class<?> playerConnection = getNMSClass("server.network.PlayerConnection");

        MethodHandles.Lookup lookup = MethodHandles.lookup();
        MethodHandle sendPacket = null;
        MethodHandle getHandle = null;
        MethodHandle connection = null;
        try {
            connection = lookup.findGetter(entityPlayer, "b", playerConnection);
            getHandle = lookup.findVirtual(craftPlayer, "getHandle", MethodType.methodType(entityPlayer));
            sendPacket = lookup.findVirtual(playerConnection, "sendPacket", MethodType.methodType(void.class, getNMSClass("network.protocol.Packet")));
        } catch (NoSuchMethodException | NoSuchFieldException | IllegalAccessException ex) {
            ex.printStackTrace();
        }

        PLAYER_CONNECTION = connection;
        SEND_PACKET = sendPacket;
        GET_HANDLE = getHandle;
    }`
But it's not a solution for multiple version support. Thought I would post this in hope it helps. My only thought to make it work on lower versions was to check the version when defining the string variable for the class when necessary, it wouldn't be very clean looking but it should work.

**Comment by LOOHP** [comment]
I thought about creating a pull request to fix this too, however I'm not sure how maintainers would like to do it tho.

**Comment by LOOHP** [comment]
Actually we do not need to check the version at all, we can jusy include both paths and try them both, since it is only looked up once, performance is not a problem.

**Comment by CryptoMorin** [comment]
Fixed in v8.1.0
